### ISSUE TITLE
Introduce requirement entry to CHANGELOG for all PRs

### ISSUE THREAD

**Issue by lilic** [prometheus-operator/prometheus-operator#3288]
As in tittle. This would make release process much faster and less error prone. 

cc @coreos/team-monitoring thoughts and 👍 👎 on this?

**Comment by metalmatze** [comment]
While it's probably helping quite a bit, keep in mind that you still most likely need to go through all PRs and check the ones that don't have added this. :man_shrugging:

**Comment by s-urbaniak** [comment]
Yes, what @metalmatze says is my experience too from other open source projects. It doesn't relieve you from iterating over all of them. It is a supportive thing for the release manager though.

**Comment by metalmatze** [comment]
Exactly. Totally worth doing, but then work sadly doesn't disappear 😁

**Comment by lilic** [comment]
Yes the point was to help out the release manager, that does not mean it won't be an automated process, but it will certainly help with the CHANGELOG text msg for example already.

**Comment by paulfantom** [comment]
I feel like this is a good idea, but after implementing it we will need to be stricter in code review and accept PRs only when they add an entry to changelog. This way we don't need any automated processes in place. I know a few projects which do this (for example home-assistant) and there is no need to iterate over all PRs when doing a release.

On the other hand modifying one file in multiple PRs can cause merge issues and contributors will be forced to rebase more frequently.

**Comment by paulfantom** [comment]
It's been some time and it would be good to arrive at some conclusion. Yay or nay on this (you can vote with emojis :+1: :-1:)?

If we want, there is a way to enforce it with CI :)

**Comment by s-urbaniak** [comment]
voted :+1: but i'd love to see the possibility to tweak the changelog before the release. I agree it is helpful though.

**Comment by brancz** [comment]
I quite like what the [google cloud platform terraform repo](https://github.com/GoogleCloudPlatform/magic-modules/pull/3956) does. It's a specifically named code block in the PR, that automatically gets collected on releases. I like this better than a changelog in the repo, as that doesn't cause conflicts, and we as maintainers can easily edit it in a PR description.

**Comment by s-urbaniak** [comment]
The latest suggestion from @brancz sgtm :+1:

**Comment by paulfantom** [comment]
> I quite like what the google cloud platform terraform repo does. It's a specifically named code block in the PR, that automatically gets collected on releases.

Sounds good to me too. There is one caveat though. That repo is using https://github.com/hashicorp/go-changelog for checking for automatic checking if changelog block is populated. Sadly this tool cannot be used directly in our CI workflow as GitHub Actions don't allow using GITHUB_TOKEN in PRs originating from forks. Which means we would need to remember about checking if the field was set. Or add a feature to run this tool in a cronjob ;) Either way changelog entry in PR template SGTM.

**Comment by brancz** [comment]
Agreed, it's better than nothing. Also, I would imagine if there was a pre-baked github action (whether it's using hashicorp/go-changelog or not), would probably be a very much used community GitHub action if anyone wants to build it.

**Comment by lilic** [comment]
So we seem to have an agreement or no one is really opposed to this? Anyone want to take this on?

**Comment by paulfantom** [comment]
Yes, I will implement it. The starting PR was already merged in #3719

**Comment by bwplotka** [comment]
I wonder if we can leverage what @simonpasquier build 🤗 

https://github.com/prometheus/promu/pull/170

NOTE: We are also looking on something that works for Thanos/Promethus, so subscribing this 🙃

**Comment by ArthurSens** [comment]
This is still desired!

**Comment by slashpai** [comment]
I actually had thought about this while doing releases, may be we could add like how Prometheus does https://github.com/prometheus/prometheus/blob/main/CHANGELOG.md#unreleased?

**Comment by ArthurSens** [comment]
It could work if we add a CI check to block if there's no changelog entry. In prometheus it's already super clunky because PRs more often don't include that entry in `Unreleased` and we get merge conflicts when merging the release branch back to main and there are entries in the `Unreleased` section :(
